_episodes/09-cmorization.md

```diff
@@ -0,0 +1,1053 @@
+> >           lat                                         -        x         -
+> >           lon                                         -        -         x
+> >      Attributes:
+> >           created_by: Fabian Gans [fgans@bgc-jena.mpg.de], Ulrich Weber [uweber@bgc-jena.mpg...
+> >           flux: GPP
+> >           forcing: CRUNCEPv6
+> >           institution: MPI-BGC-BGI
+> >           invalid_units: gC m-2 day-1
+> >           method: Artificial Neural Networks
+> >           provided_by: Martin Jung [mjung@bgc-jena.mpg.de] on behalf of FLUXCOM team
+> >           reference: Jung et al. 2016, Nature; Tramontana et al. 2016, Biogeosciences
+> >           source_file: /mnt/lustre02/work/bd0854/b309143/Data/Tier3/FLUXCOM/OBS_FLUXCOM_reana...
+> >           temporal_resolution: monthly
+> >           title: GPP based on FLUXCOM RS+METEO with CRUNCEPv6 climate
+> >           version: v1
+> > ...
+> > ```
+> {: .solution}
+{: .challenge}
+
+The error messages that we see tell us that we do have to reformat the dataset
+slightly to have it follow the CMOR standard. It seems like the variable "gpp"
+does not have the correct unit that is given in the CMOR table where it is
+[defined](https://github.com/ESMValGroup/ESMValCore/tree/master/esmvalcore/cmor/tables/cmip6/Tables).
+It also seems like the "latitude" and "longitude" coordinates have some
+problems. So let's start writing a short python script that will fix these
+problems.
+
+The first step now is to create a file in the right folder that will contain
+the short python script. The home of all CMORizer scripts for observations
+and reanalysis datasets is
+[here](https://github.com/ESMValGroup/ESMValTool/tree/master/esmvaltool/cmorizers/obs).
+Add a file with the name ``cmorize_obs_fluxcom.py`` to this folder.
+
+> ## Note
+>
+> Always, always, when modifying or creating new code for the ESMValTool
+> repositories, work on your *own, local* branch of the ESMValTool. Optimally,
+> you have forked that branch directly from the most up-to-date version of
+> the "master" branch to avoid conflicts later when you want to merge your
+> code with the "master" branch of the ESMValTool.
+>
+{: .callout}
+
+Within our python script we have to make sure that we cover three main
+aspects of the reformatting:
+- We need to be able to locate and read the data to be CMORized
+- We have to CMORize the data (fix the CMOR problems that the ESMValTool
+had so nicely pointed out for us)
+- We need to store the data with the correct filename so that the ESMValTool
+will be able to identify it later
+
+But the very first part of the CMORizing script is a header. The header
+contains information about where to obtain the data, when it was accessed
+the last time, which ESMValTool "tier" it is associated with, and more
+detailed information about the necessary downloading and processing steps.
+
+> ## Fill out the header for the "FLUXCOM" dataset
```

Comment:
These sections are not shown correctly on the tutorial webpage. Please preview the lesson locally and fix it.